samples/nrf9160/lwm2m_client/fota.rst

```diff
@@ -0,0 +1,295 @@
+The ``update`` command uploads and generates binary resource IDs automatically when a binary file is used.
+The update default task timeout is 800 seconds.
+
+Following is an example of updating a modem instance by giving a binary file:
+
+.. code-block:: console
+
+   ./scripts/fota.py -id urn:imei:351358811331351 update 1 file mfw_nrf9160_update_from_1.3.4_to_1.3.4-FOTA-TEST.bin
+
+To use existing resource IDs, run the script with the following parameters:
+
+.. parsed-literal::
+
+   ./scripts/fota.py -id *device_id* update *instance_id* resource *resource_id*
+
+Following is an example of uploading a binary and updating a modem by referring to the uploaded resource ID:
+
+.. code-block:: console
+
+   ./scripts/fota.py upload 1 mfw_nrf9160_update_from_1.3.4-FOTA-TEST_to_1.3.4.bin
+   ./scripts/fota.py -id urn:imei:351358811331351 update 1 resource lwm2m_client_fota_instance_1
+
+
+Testing advanced FOTA
+**********************
+
+Complete the following steps to test the advanced FOTA firmware update with the lwM2M client sample and the :file:`/scripts/fota.py` file.
+
+
+   #. Download the latest released modem zip file from `nRF9160 DK Downloads`_.
+   #. Update the modem firmware using the nRF Programmer app of `nRF Connect for Desktop`_.
+   #. Copy the binaries with the following naming format from the zip file to the folder :file:`/nrf/samples/nrf9160/lwm2m_client`:
+
+      * :file:`mfw_nrf9160_update_from_1.x.x_to_1.x.x-FOTA-TEST.bin`
+      * :file:`mfw_nrf9160_update_from_1.x.x-FOTA-TEST_to_1.x.x.bin`
+
+      where 1.x.x is the latest modem release version.
+
+   #. To set up the script, you must set the username and password that you used in AVSystem's Coiote Device Management server as the environment variables.
+
+      .. code-block:: console
+
+         # Setup phase
+         export COIOTE_PASSWD='my-password'
+         export COIOTE_USER='my-username'
+
+   #. To use the :ref:`lwm2m_client` sample after updating the firmware, you must build the sample with the following overlays:
+
+         * :file:`overlay-avsystems.conf`
+         * :file:`overlay-lwm2m-1.1.conf`
+         * :file:`overlay-fota_helper.conf`
+         * :file:`overlay-adv-firmware.conf`
+
+   #. Register your device with the Coiote Device management server.
+   #. Flash the compiled sample using the erase flash option.
+   #. Wait for the device registration to be complete.
+   #. Open the :file:`src/prj.conf` file.
+   #. Change :kconfig:option:`CONFIG_MCUBOOT_IMAGE_VERSION` to ``1.0.1`` and rebuild the sample.
+   #. Update the application and modem firmware by using the :file:`/scripts/fota.py` script:
+
+      .. code-block:: console
+
+         ./scripts/fota.py -id urn:imei:351358811331351 update 0 file app_update.bin update 1 file mfw_nrf9160_update_from_1.3.4_to_1.3.4-FOTA-TEST.bin
+
+   #. Following is an example output of the command:
```

Comment:
This is not a step or a user action and hence can be part of the previous step.

Reply:
Ok I will fix this.